app/database.js

```diff
@@ -86,41 +220,89 @@ define(['sqlite', 'app/config', 'jquery', 'app/lang'], function (sqlite, config,
 			fs.writeFileSync(config.constant("DATABASE_USER"), buffer);
 			console.log("Closed connection");
 		},
-		executeQuery : function (queryname, args) {
+		executeQuery : function (queryname, args, local = true, remote = true, callback = false) {
 			var query = queries[queryname] ;
-			db.run(query, args, function(e,d) {
-				console.log(e);
-				console.log(d);
-			});
+			if (local) {
+				db.run(query, args, function(err) {
+					onError(err);
+				});
+			}
+			if (remote && database.online()) {
+				if(db_online===undefined){
+					initOnlineDB();
+				}
+				db_online.query(query, args, function(err, result) {
+					onError(err);
+					if (callback) {
+						callback();
+					}
+				});
+			} else {
+				if (callback) {
+					callback();
+				}
+			}
 		},
-		getQuery: function(queryname,args){
-			var queryResult = [];
+		getQuery: function(queryname, args) {
 			var query = queries[queryname];
+			var queryResult = [];
 			db.each(query,args, function(row, err) {
+				onError(err);
 				queryResult.push(row);
 			});
 			return queryResult;
 		},
+		getOnlineQuery: function(queryname, args, callback) {
+			if(db_online===undefined){
+				initOnlineDB();
+			}
+			var query = queries[queryname];
+			db_online.query(query, args, function(err, rows, fields) {
+				onError(err);
+				callback(rows);
+			});
+		},
 		getUnique: function(queryname, unique_name1, unique_name2, args) {
 			var queryResult = [];
 			var query = queries[queryname] ;
-			db.each(query,args, function(row, err) {
+			db.each(query, args, function(row, err) {
+				onError(err);
 				if (isUnique(unique_name1, unique_name2, queryResult, row))
 					queryResult.push(row);
 			});
 			return queryResult;
 		},
-		lastInsertRowId: function(table,row_id) {
-			var query = "SELECT "+row_id+" FROM "+table+ " ORDER BY "+row_id+ " DESC LIMIT 1";
-			db.each(query,"", function(row, err) {
+		lastInsertRowId: function(table_name, row_id) {
+			var query = "SELECT "+row_id+" FROM "+table_name+" ORDER BY "+row_id +" DESC LIMIT 1";
+			var queryResult;
+			db.each(query, function(row, err) {
 				queryResult = row[row_id];
 			});
 			return queryResult;
 		},
+		lastInsertIdOnline: function(table_name, row_id, callback) {
+			if(db_online===undefined){
+				initOnlineDB();
+			}
+			var query = "SELECT "+row_id+" FROM "+table_name+" ORDER BY "+row_id+" DESC LIMIT 1";
+			db_online.query(query, function(err, rows, fields) {
+				onError(err);
+				callback(rows[0][row_id]);
+			});
+		},
 		each : function(queryname, args, func) {
 			var query = queries[queryname];
 			db.each(query,args, func);
+		},
+		synchronize : function(userId, callback){
+			if(db_online===undefined){
```

Comment:
You check at the start of nearly every function if you need to init an online DB. Maybe go for a initOnlineDBIfRequired(), that feels cleaner to me.